app/src/pages/Edit.tsx

```diff
@@ -0,0 +1,214 @@
+import { LoaderFunction, useBlocker } from "react-router";
+import { atom } from "jotai";
+import { branchExists, createPr, repoRaw, request } from "@/api";
+import { getAtom, setAtom } from "@/App";
+import { calcCompletion, convert, isEdited, revert } from "@/data/data";
+import lessons from "@/data/lessons.json";
+import { _Entry, _Timings, Entry } from "@/data/types";
+import Footer from "@/pages/edit/Footer";
+import Header from "@/pages/edit/Header";
+import Section from "@/pages/edit/Section";
+
+/** translation edit page */
+function Edit() {
+  /** protect users from losing work */
+  useBlocker(() => !window.confirm(navMessage));
+
+  return (
+    <>
+      <Header />
+      <main>
+        <Section label="title" entries={title} />
+        <Section label="captions" entries={captions} />
+        <Section label="description" entries={description} />
+      </main>
+      <Footer />
+    </>
+  );
+}
+
+export default Edit;
+
+/** page-wide state */
+
+/** whether lesson is locked */
+export const locked = atom(false);
+/** video id */
+export const video = atom("");
+/** lesson metadata */
+export const meta = atom<(typeof lessons)[number] | null>(null);
+/** lesson language */
+let language = "";
+/** lesson title */
+export const title = atom<Entry[]>([]);
+/** lesson captions */
+export const captions = atom<Entry[]>([]);
+/** lesson description */
+export const description = atom<Entry[]>([]);
+/** selected filter id */
+export const filter = atom<Filter>("all");
+/** whether header/footer are sticky */
+export const sticky = atom(true);
+/** whether to show legacy translations */
+export const showLegacy = atom(true);
+/** completion of current lesson */
+export const completion = atom((get) =>
+  calcCompletion(get(captions), language),
+);
+
+/** filter options */
+export const filters = [
+  { id: "all", label: "All" },
+  { id: "original", label: "Original AI-generated" },
+  { id: "human", label: "Human-edited" },
+  { id: "my", label: "My edits" },
+  // "Heavily edited captions",
+  // "Recently edited captions",
+] as const;
+
+/** filter option id enum */
+type Filter = (typeof filters)[number]["id"];
+
+/** take in a entry and return whether it should be kept */
+type FilterFunc = (value: Entry) => boolean;
+
+/** filter function for each filter option */
+export const filterFuncs: Record<Filter, FilterFunc> = {
+  all: () => true,
+  original: (entry) => entry.reviews === 0,
+  human: (entry) => entry.reviews > 0,
+  my: (entry) => isEdited(entry) || entry.upvoted,
+};
+
+/** page data loader */
```

Comment:
Function that is run before a route (page) loads, to get the data the page needs. In my testing today, this sometimes took a few seconds, so when clicking on a lesson from the homepage there was some lag. Might need to add a loading indicator for that. Or I may have been rate limited by my many API calls today.